### PR TITLE
[Bug Fix] Fix the demo configuration script and remove the admin credential from `internal_user.yml` 

### PR DESCRIPTION
**Pull request by RyanL1997**
### Description
By working with @DarshitChanpura for addressing the following issue: 
> https://github.com/opensearch-project/opensearch-build/issues/4094#issuecomment-1745785846

We need the initiation of the cluster to be failed once we encountered a failure of our demo configuration script. This PR introduces the method that if we failed to set up a admin credential during the set up of demo configuration script, it will lead to an absent of admin credential section in `internal_users.yml`, so that the initiation of the cluster will be failed.

* Category (Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation)
Bug fix
* What is the old behavior before changes and new behavior after changes?
The cluster initiation will be succeed and keep using the `admin:admin` credential even if the there is an hard exit on the set up on demo configuration script.

### Issues Resolved
* Relate https://github.com/opensearch-project/opensearch-build/issues/4094
* Relate https://github.com/opensearch-project/security/pull/3329

### Testing
CI

### Check List
- [x] New functionality includes testing
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by RyanL1997** [comment]
Note: the BWC failure is expected due to the release related version changes.

File: tools/install_demo_configuration.bat [hunk 1]
```diff
@@ -359,17 +359,16 @@ if errorlevel 1 (
...
-)
+:: Append the admin credential to the end of the internal_users.yml file
+echo admin: >> !INTERNAL_USERS_FILE!
+echo   hash: "!HASHED_ADMIN_PASSWORD!" >> !INTERNAL_USERS_FILE!
+echo   reserved: true >> !INTERNAL_USERS_FILE!
+echo   backend_roles: >> !INTERNAL_USERS_FILE!
+echo   - "admin" >> !INTERNAL_USERS_FILE!
+echo   description: "Demo admin user" >> !INTERNAL_USERS_FILE!
+
+echo Content appended to !INTERNAL_USERS_FILE!
```
Comment (DarshitChanpura) [inline tools/install_demo_configuration.bat:370]: nit: Make sure the comments are same in both scripts.

Comment (RyanL1997) [reply tools/install_demo_configuration.bat]: Fixed

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -52,6 +58,33 @@ jobs:
     - name: Checkout security
       uses: actions/checkout@v4
 
+    - name: Insert Admin Credential on Linux
```
Comment (peternied) [inline .github/workflows/ci.yml:61]: What is broken without this section?

File: config/internal_users.yml [hunk 3]
```diff
@@ -10,13 +10,6 @@ _meta:
 
 ## Demo users
 
-admin:
```
Comment (peternied) [inline config/internal_users.yml:13]: Does the cluster fail to startup with this change?  Could you include what that looks like?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (RyanL1997) [reply .github/workflows/ci.yml:61]: Based on the [log](https://github.com/opensearch-project/security/actions/runs/6399851879/job/17372609229?pr=3449#step:5:7183):
```
 - org.opensearch.security.InitializationIntegrationTests.testDefaultConfig
 - org.opensearch.security.InitializationIntegrationTests.testInvalidDefaultConfig
 - org.opensearch.security.TransportUserInjectorIntegTest.testSecurityUserInjection
 - org.opensearch.security.SlowIntegrationTests.testDelayInSecurityIndexInitialization
 ```
 For example, the test of  `InitializationIntegrationTests.testDefaultConfig` is using the admin admin credential to send out the request: https://github.com/opensearch-project/security/blob/93f79f881a718f35596dcd44e705b29b8c5c25c3/src/test/java/org/opensearch/security/InitializationIntegrationTests.java#L287

**Comment by DarshitChanpura** [comment]
We do not want a cluster to start without admin credentials. What do you think about closing this @RyanL1997 ?

**Comment by peternied** [comment]
@RyanL1997 Since the password change was reverted in 2.11 - lets regroup and figure out how to approach this.  I'm going to close out this PR - we can reopen if we need it.  Thanks

**Pull request closed without merging**